src/ompl/geometric/planners/rrt/src/RRTstar.cpp

```diff
@@ -338,16 +394,15 @@ ompl::base::PlannerStatus ompl::geometric::RRTstar::solve(const base::PlannerTer
                 }
             }
 
-            if (prune_)
+            if (useNewStateRejection_)
             {
-                if (opt_->isCostBetterThan(costToGo(motion, false), bestCost_))
+                if (opt_->isCostBetterThan(solutionHeuristic(motion, false), bestCost_))
                 {
                     nn_->add(motion);
                     motion->parent->children.push_back(motion);
                 }
                 else // If the new motion does not improve the best cost it is ignored.
                 {
-                    --statesGenerated;
```

Comment:
This variable is only used in the `OMPL_INFORM` line at the end of solve as `[...] Created %u new states [...]`. 

I guess what is the interpretation of the phrase creating a new state?